backend/src/impl/benchmark_utils.py

```diff
@@ -151,36 +157,53 @@ def aggregate_view(
     ) -> pd.DataFrame:
         output_df = input_df.copy()
         for operation in view_spec.operations:
+            group_by = ["system_name"] + operation.get("group_by", [])
             if operation["op"] == "mean":
-                output_df = output_df.groupby(["system_name"]).mean()
-            if operation["op"] == "multiply":
+                output_df = output_df.groupby(group_by).mean()
+            elif operation["op"] == "multiply":
                 output_df["score"] = output_df["score"] * output_df[operation["other"]]
-            if operation["op"] == "sum":
-                output_df = output_df.groupby(["system_name"]).sum()
+            elif operation["op"] == "sum":
+                output_df = output_df.groupby(group_by).sum()
+            elif operation["op"] == "weighted_sum":
+                output_df["score"] = output_df["score"] * output_df[operation["weight"]]
```

Comment:
I wonder how to define the weights of `task`, `dataset` or `subdataset`, which is something I care about.

Reply:
My suggestion would be this (haven't had time to incorporate/test it):

```python
elif operation["op"] == "weighted_sum":
    if "weight_map" in operation:
        output_df["score"] = output_df["score"] * output_df[operation["weight"]].map(operation["weight_map"])
    else:
        output_df["score"] = output_df["score"] * output_df[operation["weight"]]
```

So then you could do something like this:
```python
"operations" = [
    {"op": "weighted_sum", "weight": "dataset", "weight_map": {"snli": 0.7, "sst2": 0.3}}
]
```

Does that look easy enough to use?

Reply:
Thanks for the reply!
I think the first option (using the `map`) is clearly better!
While another confusing point comes out:
When will we use the `option['other']`, which seems to be triggered by 
`elif operation["op"] == "multiply":`

Do we need to modify something [here](https://github.com/neulab/explainaboard_web/blob/benchmark_alternative_schema/backend/src/impl/benchmark_configs/config_miniglue.json)?